### DISCUSSION TITLE
No modal window with services

### DISCUSSION THREAD

**Discussion by waynelogic** [OFFLINE-GmbH/oc-mall-plugin#953] [General]
When i add any services to my product, according to the author's idea, when i press add the onAddToCart method is initialized, a modal window with services is renderind with a form with **onAddToCartWithServices** method. In my case, this method is not suitable. My services should be listed right under my custom fields. But when using the **onAddToCartWithServices** method on the main cart form, then Custom Fields do not fall into the cart. What to do in this situation?
![Screenshot_5](https://user-images.githubusercontent.com/90955971/189573566-2eda11ca-f0db-4c81-ac62-8bd592d81371.png)
Red area is custom fields, green - services.

**Comment by tobias-kuendig** [comment]
If the basic setup of the component does not suit your use-case, you can always extend the component and override the methods, or even create a completely new component that does exactly what you need.
